src/helm-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}
```

Comment:
Generally, I put the chart name here. For that: `.Chart.Name`

Reply:
In fact, customizing the deployment name makes sense when we want to deploy different kinds of PaperMC servers in a same namespace.
So I would go over something like `default .Chart.Name .Values.name`.
More details: https://helm.sh/docs/chart_template_guide/function_list/#default

Reply:
Seems like you need to inject the value multiple times in templates.

A first idea is to go over variables: https://helm.sh/docs/chart_template_guide/variables/#helm.
That's a good idea when you want to remove duplications across a single template.

But in your case, you need this logic to be defined at both deployment and service levels (so across multiple templates).
This is why I recommend you instead to rely on "Named templates": https://helm.sh/docs/chart_template_guide/named_templates/#helm